ledger/acctonline.go

```diff
@@ -0,0 +1,846 @@
+	// entries when all changes to an account have been reflected in
+	// the account DB, and no outstanding modifications remain.
+	ndeltas int
+}
+
+// cachedOnlineAccount is a light-weight version of persistedOnlineAccountData suitable for in-memory caching
+//msgp:ignore cachedOnlineAccount
+type cachedOnlineAccount struct {
+	baseOnlineAccountData
+	updRound basics.Round
+}
+
+// onlineAccounts tracks history of online accounts
+type onlineAccounts struct {
+	// Connection to the database.
+	dbs db.Pair
+
+	// Prepared SQL statements for fast accounts DB lookups.
+	accountsq *onlineAccountsDbQueries
+
+	// cachedDBRoundOnline is always exactly tracker DB round (and therefore, onlineAccountsRound()),
+	// cached to use in lookup functions
+	cachedDBRoundOnline basics.Round
+
+	// deltas stores updates for every round after dbRound.
+	deltas []ledgercore.AccountDeltas
+
+	// accounts stores the most recent account state for every
+	// address that appears in deltas.
+	accounts map[basics.Address]modifiedOnlineAccount
+
+	// onlineRoundParamsData stores onlineMoney, rewards from rounds
+	// dbRound + 1 - maxLookback to current round, where maxLookback is max(proto.MaxBalLookback, votersLookback)
+	// It behaves as delta storage and a cache.
+	onlineRoundParamsData []ledgercore.OnlineRoundParamsData
+
+	// log copied from ledger
+	log logging.Logger
+
+	// ledger is the source ledger, which is used to synchronize
+	// the rounds at which we need to flush the balances to disk
+	// in favor of the catchpoint to be generated.
+	ledger ledgerForTracker
+
+	// deltasAccum stores the accumulated deltas for every round starting dbRound-1.
+	deltasAccum []int
+
+	// accountsMu is the synchronization mutex for accessing the various non-static variables.
+	accountsMu deadlock.RWMutex
+
+	// accountsReadCond used to synchronize read access to the internal data structures.
+	accountsReadCond *sync.Cond
+
+	// voters keeps track of Merkle trees of online accounts, used for compact certificates.
+	voters *votersTracker
+
+	// baseAccounts stores the most recently used accounts, at exactly dbRound
+	baseOnlineAccounts lruOnlineAccounts
+
+	// onlineAccountsCache contains up to onlineAccountsCacheMaxSize accounts with their complete history
+	// for the range [Lastest - MaxBalLookback - X, Latest - lookback], where X = [0, commit range]
+	// and alway containing an entry for Lastest - MaxBalLookback + 1 if some account is cached.
+	// The invariant is held by
+	// 1) loading a full history when new accounts get added
+	// 2) adding online accounts state changes when flushing to disk
+	// 3) pruning the history by removing older than Lastest - MaxBalLookback non-online entries
+	onlineAccountsCache onlineAccountsCache
+
+	// maxAcctLookback sets the minimim deltas size to keep in memory
+	acctLookback uint64
+}
+
+// initialize initializes the accountUpdates structure
+func (ao *onlineAccounts) initialize(cfg config.Local) {
```

Comment:
Better to pass in the specific field that is required instead of the entire `config.Local` object.

Reply:
I kind of agree but other trackers need 3+ options and their `initialize` accepts cfg